src/handshake/server.rs

```diff
@@ -37,6 +36,13 @@ pub struct Server<'a, T> {
 	buffer: BytesMut,
 }
 
+/// Server configuration can't be generated by hand. Using the `http` feature, this is returned
+/// from `crate::handshake::http::upgrade_request`, and can be handed to [`Server::configure`]
+/// in order to configure the server instance appropriately based on the request headers.
```

Comment:
I think it'd be good to be clear that this is only about enabling/disabling the deflate extension, so nobody thinks they can configure their whole server using http headers. Maybe it could be named `ExtensionConfig` even?